backend/fms_core/services/process.py

```diff
@@ -0,0 +1,22 @@
+from django.core.exceptions import ValidationError
+from fms_core.models import Process
+
+def create_processes_for_experiment_from_protocols_dict(protocols_objs_dict,
```

Comment:
Could this simply be create_process_hierarchy_from_protocol ? This may be more inviting to reuse it. I'm not sure about the top process naming ...

Reply:
You didnt like the parent process, I suggest top process unless you have a better idea.

Dict is more clear that were not dealing with straight up objects